components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/index.tsx

```diff
@@ -47,42 +47,52 @@ function SearchSummary (props: { searchQueries: string[] }) {
   )
 }
 
+function AssistantEvent(props: { event: Mojom.ConversationEntryEvent, hasCompletionStarted: boolean, isEntryInProgress: boolean }) {
+  if (props.event.completionEvent) {
+    return (
+      <MarkdownRenderer
+        shouldShowTextCursor={props.isEntryInProgress}
+        text={props.event.completionEvent.completion}
+      />
+    )
+  }
+  if (props.event.searchStatusEvent && props.isEntryInProgress && !props.hasCompletionStarted) {
+    return (
+      <div className={styles.actionInProgress}><ProgressRing />Improving answer with Brave Search…</div>
+    )
+  }
+  if (props.event.pageContentRefineEvent && props.isEntryInProgress && !props.hasCompletionStarted) {
+    return (
+      <div className={styles.actionInProgress}><ProgressRing />{getLocale('pageContentRefinedInProgress')}</div>
+    )
+  }
+  // TODO(petemill): Consider displaying in-progress queries if the API
+  // timing improves (or worsens for the completion events).
+  // if (event.searchQueriesEvent && props.isEntryInProgress) {
+  //   return (<>
+  //     {event.searchQueriesEvent.searchQueries.map(query => <div className={styles.searchQuery}>Searching for <span className={styles.searchLink}><Icon name="brave-icon-search-color" /><Link href='#'>{query}</Link></span></div>)}
+  //   </>)
+  // }
+
+  // Unknown events should be ignored
+  return null
+}
+
 export default function AssistantResponse(props: { entry: Mojom.ConversationTurn, isEntryInProgress: boolean }) {
   const searchQueriesEvent = props.entry.events?.find(event => !!event.searchQueriesEvent)?.searchQueriesEvent
-  const hasCompletionStarted = !props.isEntryInProgress || props.entry.events?.find(event => !!event.completionEvent)
+  const hasCompletionStarted = !props.isEntryInProgress ||
+    (props.entry.events?.some(event => !!event.completionEvent) ?? false)
```

Comment:
superfluous `!!`
```suggestion
    (props.entry.events?.some(event => event.completionEvent) ?? false)
```

optional:

```suggestion
const hasCompletionStarted = props.isEntryInProgress
    || !!props.entry.events?.some(e => e.completionEvent)
```

Reply:
I've been studying this for a good few minutes and I still can't quickly reason with the names here, or how your suggestion is equivalent to what's there at the moment. Not introduced in this PR but I think we can improve it at some point.